cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -1490,6 +1578,9 @@ void TestBackpressure(BatchesMaker maker, int num_batches, int batch_size,
     std::string name() const { return name_prefix + ";" + (is_fast ? "fast" : "slow"); }
   };
 
+  Gate gate;
+  GatedNodeOptions gate_options(&gate);
```

Comment:
I don't see `gate_options` is used anywhere. @westonpace, are you sure the tester is doing what you intended?

Reply:
Good catch.  I must be missing something or getting lucky with timing